extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -51,7 +51,8 @@
 class KubernetesProcessor {
 
     private static final String PROPERTY_PREFIX = "dekorate.";
-    private static final String ALLOWED_GENERATOR = "kubernetes";
+    private static final List<String> ALLOWED_GENERATORS = Arrays.asList("kubernetes", "openshift", "knative", "docker", "s2i");
```

Comment:
How about making this a `Set` so `contains` will be O(1)?

Reply:
Done